apps/game/lib/game/cup.ex

```diff
@@ -0,0 +1,12 @@
+defmodule Cup do
+  @moduledoc """
+  generat a hand of n dice in a cup
+  """
+
+  @spec new(integer) :: [integer]
+  def new(n, acc \\ [])
+  def new(0, acc), do: acc
+  def new(n, acc) do
+    new(n-1, [:rand.uniform(6) | acc])
+  end
```

Comment:
I've got another alternative... :)

This is the opposite of a `reduce` (ie a reduce takes a list and reduces it to a value, this one takes a function and builds a list out of it).

Anyway, you could use `Stream.repeatedly/1` and `Enum.take` to build a list ... this would be an incredibly concise, elegant, clear way to build this function IMO.

https://hexdocs.pm/elixir/Stream.html#repeatedly/1